builder/simplify.cc

```diff
@@ -409,40 +473,48 @@ class simplifier : public node_mutator {
+        k.push_back(set_value_in_scope(info_map, v->sym, {point(b), al}));
       }
       if (const variable* v = b.as<variable>()) {
         // bounds of b are [a, a].
-        k.push_back(set_value_in_scope(bounds, v->sym, point(a)));
+        const std::optional<expr_info>& old_info = info_map[v->sym];
+        const alignment_type& al = old_info ? old_info->alignment :  alignment_type();
+        k.push_back(set_value_in_scope(info_map, v->sym, {point(a),  al}));
       }
     }
 
     void learn_from_less(const expr& a, const expr& b) {
       if (const variable* v = a.as<variable>()) {
         // a has an upper bound of b - 1
-        const std::optional<interval_expr>& old_bounds = bounds[v->sym];
-        const expr& lb = old_bounds ? old_bounds->min : expr();
-        k.push_back(set_value_in_scope(bounds, v->sym, {lb, b - 1}));
+        const std::optional<expr_info>& old_info = info_map[v->sym];
+        const expr& lb = old_info ? old_info->bounds.min : expr();
+        const alignment_type& al = old_info ? old_info->alignment :  alignment_type();
+        k.push_back(set_value_in_scope(info_map, v->sym, {{lb, b - 1}, al}));
       }
       if (const variable* v = b.as<variable>()) {
         // b has a lower bound of a + 1
-        const std::optional<interval_expr>& old_bounds = bounds[v->sym];
-        const expr& ub = old_bounds ? old_bounds->max : expr();
-        k.push_back(set_value_in_scope(bounds, v->sym, {a + 1, ub}));
+        const std::optional<expr_info>& old_info = info_map[v->sym];
+        const expr& ub = old_info ? old_info->bounds.max : expr();
+        const alignment_type& al = old_info ? old_info->alignment :  alignment_type();
+        k.push_back(set_value_in_scope(info_map, v->sym, {{a + 1, ub}, al}));
       }
     }
     void learn_from_less_equal(const expr& a, const expr& b) {
       if (const variable* v = a.as<variable>()) {
         // a has an upper bound of b
-        const std::optional<interval_expr>& old_bounds = bounds[v->sym];
-        const expr& lb = old_bounds ? old_bounds->min : expr();
-        k.push_back(set_value_in_scope(bounds, v->sym, {lb, b}));
+        const std::optional<expr_info>& old_info = info_map[v->sym];
+        const expr& lb = old_info ? old_info->bounds.min : expr();
+        const alignment_type& al = old_info ? old_info->alignment :  alignment_type();
+        k.push_back(set_value_in_scope(info_map, v->sym, {{lb, b}, al}));
       }
       if (const variable* v = b.as<variable>()) {
         // b has a lower bound of a
-        const std::optional<interval_expr>& old_bounds = bounds[v->sym];
-        const expr& ub = old_bounds ? old_bounds->max : expr();
-        k.push_back(set_value_in_scope(bounds, v->sym, {a, ub}));
+        const std::optional<expr_info>& old_info = info_map[v->sym];
+        const expr& ub = old_info ? old_info->bounds.max : expr();
+        const alignment_type& al = old_info ? old_info->alignment :  alignment_type();
```

Comment:
I think there should be a helper function for this sequence of 3 lines repeated several times (also since it needs changing maybe to fix the reference issue).

Reply:
I added a helper function for getting alignment (the bounds lines are slightly different) and changed it to non-refernce.